### PR TITLE
Replaces /datum/status_effect/crusher_damage with /datum/component/crusher_damage

### PR DESCRIPTION
**Pull request by IndusRobot**
## About The Pull Request

Replaces status_effect/crusher_damage with component/crusher_damage

## Why It's Good For The Game

Theoretically better performance for the same functionality, although I'm not sure if a component was the right choice considering it's so tiny now. I verified crusher trophies were still dropping correctly afterwards. 

## Changelog
:cl:
fix: Replaced /datum/status_effect/crusher_damage with /datum/component/crusher_damage
/:cl:

### CODE REVIEW

**Comment by tmtmtl30** [comment]
Yeah, this is more or less what I had in mind. These were clogging up SSfastprocess, since one is created every time an asteroid hostilemob or megafauna spawns, and we have a whole fuckin' lot of those. i suggested it be moved to a component and out of status effects to make the level of modularity roughly the same, and because that was more or less a drop-in change.

the code is still Not Great, though -- crusher damage tracking, along with crusher_loot, could easily be moved to a variable on /mob/living/simple_animal/hostile, since both megafauna and asteroid hostiles are derived from that. that would probably make quite a bit more sense. but this gets the job done at least

**Comment by tmtmtl30** [comment]
i switched capitalization modes after the second sentence of that comment. fascinating

**Review by Fikou — CHANGES_REQUESTED** [review]
GetComponent is not a great solution. i suggest a global associated list of mob = damage

File: code/modules/mining/equipment/kinetic_crusher.dm [hunk 1]
```diff
@@ -82,15 +82,14 @@
 		to_chat(user, "<span class='warning'>[src] is too heavy to use with one hand! You fumble and drop everything.</span>")
 		user.drop_all_held_items()
 		return
-	var/datum/status_effect/crusher_damage/C = target.has_status_effect(STATUS_EFFECT_CRUSHERDAMAGETRACKING)
+	var/datum/component/crusher_damage/crusher_damage = target.GetComponent( /datum/component/crusher_damage )
 	var/target_health = target.health
 	..()
-	for(var/t in trophies)
+	for(var/obj/item/crusher_trophy/trophy in trophies)
```
Comment (Fikou) [inline code/modules/mining/equipment/kinetic_crusher.dm:88]: as anything

File: code/modules/mining/equipment/kinetic_crusher.dm [hunk 2]
```diff
@@ -118,25 +117,24 @@
 		var/datum/status_effect/crusher_mark/CM = L.has_status_effect(STATUS_EFFECT_CRUSHERMARK)
 		if(!CM || CM.hammer_synced != src || !L.remove_status_effect(STATUS_EFFECT_CRUSHERMARK))
 			return
-		var/datum/status_effect/crusher_damage/C = L.has_status_effect(STATUS_EFFECT_CRUSHERDAMAGETRACKING)
+		var/datum/component/crusher_damage/crusher_damage = L.GetComponent( /datum/component/crusher_damage )
 		var/target_health = L.health
-		for(var/t in trophies)
-			var/obj/item/crusher_trophy/T = t
-			T.on_mark_detonation(target, user)
+		for(var/obj/item/crusher_trophy/trophy in trophies)
```
Comment (Fikou) [inline code/modules/mining/equipment/kinetic_crusher.dm:122]: as anything

File: code/modules/mining/equipment/kinetic_crusher.dm [hunk 3]
```diff
@@ -843,15 +841,14 @@
 		to_chat(user, "<span class='warning'>[src] is too heavy to use with one hand! You fumble and drop everything.</span>")
 		user.drop_all_held_items()
 		return
-	var/datum/status_effect/crusher_damage/C = target.has_status_effect(STATUS_EFFECT_CRUSHERDAMAGETRACKING)
+	var/datum/component/crusher_damage/crusher_damage = target.GetComponent( /datum/component/crusher_damage )
 	var/target_health = target.health
 	..()
-	for(var/t in trophies)
+	for(var/obj/item/crusher_trophy/trophy in trophies)
```
Comment (Fikou) [inline code/modules/mining/equipment/kinetic_crusher.dm:847]: as anything

File: code/modules/mining/equipment/kinetic_crusher.dm [hunk 4]
```diff
@@ -879,25 +876,24 @@
 		var/datum/status_effect/crusher_mark/CM = L.has_status_effect(STATUS_EFFECT_CRUSHERMARK)
 		if(!CM || CM.hammer_synced != src || !L.remove_status_effect(STATUS_EFFECT_CRUSHERMARK))
 			return
-		var/datum/status_effect/crusher_damage/C = L.has_status_effect(STATUS_EFFECT_CRUSHERDAMAGETRACKING)
+		var/datum/component/crusher_damage/crusher_damage = L.GetComponent( /datum/component/crusher_damage )
 		var/target_health = L.health
-		for(var/t in trophies)
-			var/obj/item/crusher_trophy/T = t
-			T.on_mark_detonation(target, user)
+		for(var/obj/item/crusher_trophy/trophy in trophies)
```
Comment (Fikou) [inline code/modules/mining/equipment/kinetic_crusher.dm:881]: as anything

**Pull request closed without merging**